V3 boards list

alexkorsun opened the pull request.

alexkorsun committed:
- wip

alexkorsun committed:
- wip

alexkorsun committed:
Merge remote-tracking branch 'origin/v3' into v3-boards-list

alexkorsun committed:
- fixed task-team component issues (removed user becomes null, wrong usage of filter, map)

alexkorsun committed:
- fixed issue with items in baskets (API endpoint)
- updated createTask method to set root task
- added app to container

alexkorsun committed:
- updated baskets view
- fixed content of developers array

alexkorsun committed:
- back fixtures loading

alexkorsun committed:
- updated gitignore

alexkorsun committed:
- cleaned up

alexkorsun committed:
- updated migrate.js

alexkorsun committed:
- updated nav with dropdown for baskets

alexkorsun committed:
- fixed migration for tasks

irinadrozdova commented on app/api/user/index.js:
```diff
@@ -1,41 +1,92 @@
+    let Task = app.container.get('Task');
+
+    app.get('/api/users/:userId', function (req, res, next) {
+        User.findById(req.params.userId, '-local.passwordHashed -local.passwordSalt', function (err, user) {
```
может это стоит двинуть в UserService.getUser()

irinadrozdova commented on app/api/user/index.js:
```diff
@@ -1,41 +1,92 @@
+                        return next(err);
+                    }
+
+                    res.json(users);
```
1. чуток не понятно, почему по роуту мы должны забрать projects,  а отдаем юзеров 
2. давай двинем это в ТаскСервис в метод, который будет возвращать всех моих коллег по проектам ))

irinadrozdova commented on migrations/001-add-root-to-tasks.js:
```diff
@@ -0,0 +1,50 @@
+let Task = app.container.get('Task');
+let TaskService = app.container.get('TaskService');
+
+exports.up = function (next) {
```
давай стрелочную функцию ))

irinadrozdova commented on migrations/001-add-root-to-tasks.js:
```diff
@@ -0,0 +1,50 @@
+        .pipe(es.map(function (task, next) {
+            ps.pause();
+
+            TaskService.getRoot(task, function (err, root) {
```
давай стрелочную функцию ))

irinadrozdova commented on migrations/001-add-root-to-tasks.js:
```diff
@@ -0,0 +1,50 @@
+
+                let rootId = task.parentTaskId ? root._id : null;
+
+                Task.update({_id: task._id}, {$set: {root: rootId}}, function (err) {
```
давай стрелочную функцию ))

irinadrozdova commented on public-v2/src/app/tracker/basket/baskets.component.ts:
```diff
@@ -0,0 +1,117 @@
+
+  onChangeDate(date): void {
+    this.showDatePicker = false;
+    this.date = date;
```
почему не написать this.date = this.prepareDate(date) ?
в любом случае this.date = date это ссылка будет и this.date  после строки date = this.prepareDate(date) изменится

irinadrozdova committed:
- updated user index.js
- updated TaskService. Added getMyColleagues
- updated UserService. Added getUserById

irinadrozdova committed:
- cleaned up

irinadrozdova committed:
Merge branch 'v3' into v3-boards-list
